packages/pds/src/app-view/services/indexing/plugins/list-item.ts

```diff
@@ -0,0 +1,99 @@
+import { AtUri } from '@atproto/uri'
+import { CID } from 'multiformats/cid'
+import * as ListItem from '../../../../lexicon/types/app/bsky/graph/listitem'
+import * as lex from '../../../../lexicon/lexicons'
+import Database from '../../../../db'
+import {
+  DatabaseSchema,
+  DatabaseSchemaType,
+} from '../../../../db/database-schema'
+import { BackgroundQueue } from '../../../../event-stream/background-queue'
+import RecordProcessor from '../processor'
+
+const lexId = lex.ids.AppBskyGraphListitem
+type IndexedListItem = DatabaseSchemaType['list_item']
+
+const insertFn = async (
+  db: DatabaseSchema,
+  uri: AtUri,
+  cid: CID,
+  obj: ListItem.Record,
+  timestamp: string,
+): Promise<IndexedListItem | null> => {
+  const inserted = await db
+    .insertInto('list_item')
+    .values({
+      uri: uri.toString(),
+      cid: cid.toString(),
+      creator: uri.host,
+      subjectDid: obj.subject,
+      listUri: obj.list,
+      reason: obj.reason,
+      reasonFacets: obj.reasonFacets
+        ? JSON.stringify(obj.reasonFacets)
+        : undefined,
+      createdAt: obj.createdAt,
+      indexedAt: timestamp,
+    })
+    .onConflict((oc) => oc.doNothing())
+    .returningAll()
+    .executeTakeFirst()
+  return inserted || null
+}
+
+const findDuplicate = async (
+  db: DatabaseSchema,
+  uri: AtUri,
+  obj: ListItem.Record,
+): Promise<AtUri | null> => {
+  const found = await db
+    .selectFrom('list_item')
+    .where('creator', '=', uri.host)
+    .where('listUri', '=', obj.list)
```

Comment:
If we validate that the creator of the list item is the creator of the list, then we can express that a subject should only appear in a list once by checking `(listUri, subjectDid)` rather than `(creator, listUri, subjectDid)`.

Reply:
ahhh yup yup okay i dig this approach 👍